### ISSUE TITLE
LogstashFormatter.isIncludeMdc() always return true

### ISSUE THREAD

**Issue by knok16** [logstash/logstash-logback-encoder#103]
Hi,
In [commit](https://github.com/logstash/logstash-logback-encoder/commit/e9de1adf7167b2855f4fadb170f48ac3e785f306) `LogstashFormatter.isIncludeMdc`/`LogstashFormatter.setIncludeMdc` logic was changed.
It's intoduced bug: even if `setIncludeMdc(false)` was called - `isIncludeMdc()` always return `true`.
[setIncludeMdc](https://github.com/logstash/logstash-logback-encoder/blob/master/src/main/java/net/logstash/logback/LogstashFormatter.java#L233) can be fixed in next way:

```
public void setIncludeMdc(boolean includeMdc) {
    if (isIncludeMdc() != includeMdc) {
        getProviders().removeProvider(mdcProvider);
        if (includeMdc) {
            getProviders().addMdc(mdcProvider = new MdcJsonProvider());
        } else {
        	mdcProvider = null;
        }
    }
}
```

Thanks

**Comment by philsttr** [comment]
Fixed for 4.4

**Comment by philsttr** [comment]
4.4 has been released with this fix
